# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1435235541 14400
#      Thu Jun 25 08:32:21 2015 -0400
# Node ID 085c99b96aed1f46da04a32e25ff23fa63b09f78
# Parent  a0fab64685ab2e61db95ecd0c211b832daf9d34f
# Parent  977853866d6969358ceb20421a54b736e3e26894
maint: Merge away extra head on default.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -1420,48 +1420,43 @@ function __go_draw_axes__ (h, plot_strea
   have_data = (! (isempty (data) || all (cellfun ("isempty", data))));
 
   ## Note we don't use the [xy]2range of gnuplot as we don't use the
   ## dual axis plotting features of gnuplot.
   if (isempty (xlim))
     return;
   endif
   if (strcmpi (axis_obj.xdir, "reverse"))
-    xdir = "reverse";
-  else
-    xdir = "noreverse";
+    xlim = flip (xlim);
   endif
-  fprintf (plot_stream, "set xrange [%.15e:%.15e] %s;\n", xlim, xdir);
+
+  fprintf (plot_stream, "set xrange [%.15e:%.15e];\n", xlim);
   if (strcmpi (axis_obj.xaxislocation, "top"))
-    fprintf (plot_stream, "set x2range [%.15e:%.15e] %s;\n", xlim, xdir);
+    fprintf (plot_stream, "set x2range [%.15e:%.15e];\n", xlim);
   endif
 
   if (isempty (ylim))
     return;
   endif
   if (strcmpi (axis_obj.ydir, "reverse"))
-    ydir = "reverse";
-  else
-    ydir = "noreverse";
+    ylim = flip (ylim);
   endif
-  fprintf (plot_stream, "set yrange [%.15e:%.15e] %s;\n", ylim, ydir);
+  fprintf (plot_stream, "set yrange [%.15e:%.15e];\n", ylim);
   if (strcmpi (axis_obj.yaxislocation, "right"))
-    fprintf (plot_stream, "set y2range [%.15e:%.15e] %s;\n", ylim, ydir);
+    fprintf (plot_stream, "set y2range [%.15e:%.15e];\n", ylim);
   endif
 
   if (nd == 3)
     if (isempty (zlim))
       return;
     endif
     if (strcmpi (axis_obj.zdir, "reverse"))
-      zdir = "reverse";
-    else
-      zdir = "noreverse";
+      zlim = flip (zlim);
     endif
-    fprintf (plot_stream, "set zrange [%.15e:%.15e] %s;\n", zlim, zdir);
+    fprintf (plot_stream, "set zrange [%.15e:%.15e];\n", zlim);
   endif
 
   cmap = parent_figure_obj.colormap;
   cmap_sz = rows (cmap);
   if (! any (isinf (clim)))
     if (truecolor || ! cdatadirect)
       if (rows (addedcmap) > 0)
         for i = 1:data_idx
